From 7a919d4edddfcf43f821c3c81a1637ac5c2df84b Mon Sep 17 00:00:00 2001
From: Adnan Begovic <adnan@cyngn.com>
Date: Thu, 10 Mar 2016 09:57:24 -0800
Subject: [PATCH 2/2] telephony: Move COLP definition to android settings.

  Otherwise we pollute the bootclasspath with public cmsdk
  symbols.

  2) testBootClassPathIsClean(org.cyanogenmod.tests.versioning.unit.ClassPathTest)
java.lang.AssertionError: Jar file /system/framework/telephony-common.jar should not have cyanogenmod.alarmclock.ClockContract$AlarmsColumns !

Change-Id: I993897480a30ca40e9adca19deb36b13b4243a97
---
 Android.mk                                                      | 1 -
 src/java/com/android/internal/telephony/GsmCdmaCallTracker.java | 2 +-
 src/java/com/android/internal/telephony/GsmCdmaConnection.java  | 7 +++----
 3 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/Android.mk b/Android.mk
index b97dba3..717a7e8 100644
--- a/Android.mk
+++ b/Android.mk
@@ -27,7 +27,6 @@ LOCAL_SRC_FILES := $(call all-java-files-under, src/java) \
 
 LOCAL_JAVA_LIBRARIES := voip-common ims-common telephony-ext
 LOCAL_STATIC_JAVA_LIBRARIES := ims-ext-common
-LOCAL_STATIC_JAVA_LIBRARIES += org.cyanogenmod.platform.internal
 
 ifneq ($(BOARD_RIL_CLASS),)
 LOCAL_SRC_FILES += $(call find-other-java-files,$(BOARD_RIL_CLASS))
diff --git a/src/java/com/android/internal/telephony/GsmCdmaCallTracker.java b/src/java/com/android/internal/telephony/GsmCdmaCallTracker.java
index 1ed4210..eacaa11 100755
--- a/src/java/com/android/internal/telephony/GsmCdmaCallTracker.java
+++ b/src/java/com/android/internal/telephony/GsmCdmaCallTracker.java
@@ -82,7 +82,7 @@ public class GsmCdmaCallTracker extends CallTracker {
     private GsmCdmaConnection mPendingMO;
     private boolean mHangupPendingMO;
 
-    private GsmCdmaPhone mPhone;
+    protected GsmCdmaPhone mPhone;
 
     private boolean mDesiredMute = false;    // false = mute off
 
diff --git a/src/java/com/android/internal/telephony/GsmCdmaConnection.java b/src/java/com/android/internal/telephony/GsmCdmaConnection.java
index 3e17232..c905494 100644
--- a/src/java/com/android/internal/telephony/GsmCdmaConnection.java
+++ b/src/java/com/android/internal/telephony/GsmCdmaConnection.java
@@ -24,6 +24,7 @@ import android.os.PersistableBundle;
 import android.os.PowerManager;
 import android.os.Registrant;
 import android.os.SystemClock;
+import android.provider.Settings;
 import android.telephony.CarrierConfigManager;
 import android.telephony.DisconnectCause;
 import android.telephony.Rlog;
@@ -37,8 +38,6 @@ import com.android.internal.telephony.uicc.UiccCardApplication;
 import com.android.internal.telephony.uicc.UiccController;
 import com.android.internal.telephony.uicc.IccCardApplicationStatus.AppState;
 
-import cyanogenmod.providers.CMSettings;
-
 /**
  * {@hide}
  */
@@ -780,8 +779,8 @@ public class GsmCdmaConnection extends Connection {
         mCnapNamePresentation = dc.namePresentation;
 
         boolean connectedLineIdentification =
-                CMSettings.System.getInt(mOwner.mPhone.getContext().getContentResolver(),
-                        CMSettings.System.CONNECTED_LINE_IDENTIFICATION, 1) != 0;
+                Settings.Global.getInt(mOwner.mPhone.getContext().getContentResolver(),
+                        Settings.Global.CONNECTED_LINE_IDENTIFICATION, 1) != 0;
         if (mIsIncoming || connectedLineIdentification)
             mNumberPresentation = dc.numberPresentation;
 
-- 
2.9.3

